### PR TITLE
Saving offsets with N+1 elements

### PR DESCRIPTION
**Pull request by frheault**
This is a test following @neurolabusc suggestion to have the TRX specifications of offsets closer to the one of VTK.
This means the offsets have N+1 elements (nb_streamlines+1) and the last one is equal to nb_vertices.

This does not change much in terms of internal use for Nibabel except we ignore the last element when loading and add the last element when saving.

This is simply a test to see what had to be changed. This also means everything testing TRX file generated would have to be modified and re-uploaded.

That was @neurolabusc suggestion and he wrote me this before:

> As an aside, I would urge the TRX specification to follow the VTK lead in solving the fence post problem for the offsets. The fact that the length of the final streamline can not be determined from the offsets alone, but must be inferred from the number of positions is awkward. Fast code really needs this to avoid the penalty of range check conditionals. Therefore, I really think storing NB_STREAMLINES + 1 offsets is correct. VTK sets a precedence for this and it makes any code supporting this format cleaner, easier to maintain and simpler to understand.

As the proposal is still up for debate and modification, I wanted to show it is easy to change/support for him. But I would leave it to everyone to decide if we change the specification.

### CODE REVIEW

**Comment by Lestropie** [comment]
Conceptually, I think that this makes sense. Implementations that are designed around tracking the total vertex count can still work just by ignoring the additional data, but this facilitates more computationally efficient implementations.And as @neurolabusc says, there's a precedent set in a far more mature format.

**Comment by arokem** [comment]
+1 to the suggested change! 

Along the way, we've run here into the issue of versioning the test data, which would fall out of spec... I think that it's actually possible to do in Figshare (based on reading this page: https://help.figshare.com/article/can-i-edit-or-delete-my-research-after-it-has-been-made-public). I think that we'd need to document somewhere in the code which version of the data we are reading.

File: trx/fetcher.py [hunk 1]
```diff
@@ -21,14 +21,14 @@ def get_testing_files_dict():
...
-        ('https://figshare.com/ndownloader/files/35709968',
-                '8861975246befac982c1f47e434c60cf'),
+        ('https://figshare.com/ndownloader/files/37624148',
+         '03f7651a0f9e3eeabee9aed0ad5f69e1'),
         'trx_from_scratch.zip':
-        ('https://figshare.com/ndownloader/files/35709764',
-                '511c59a3ee8c16b4be0ef815b889e939')
+        ('https://figshare.com/ndownloader/files/37624151',
+         'd9f220a095ce7f027772fcd9451a2ee5')
     }
```
Comment (arokem) [inline trx/fetcher.py:33]: Is there some way to record here which version of the figshare data is being used?

Comment (frheault) [reply trx/fetcher.py:33]: I added a comment at the end of each line with the version and date from figshare.

File: trx/fetcher.py [hunk 2]
```diff
@@ -21,14 +21,14 @@ def get_testing_files_dict():
     """ Get dictionary linking zip file to their Figshare URL & MD5SUM """
     return {
         'DSI.zip':
-            ('https://figshare.com/ndownloader/files/35617193',
-             '7d7082f7f2e07cb39e2fef13107af169'),
+        ('https://figshare.com/ndownloader/files/37624154',
+         'b847f053fc694d55d935c0be0e5268f7'), # V1 (27.09.2022)
```
Comment (arokem) [inline trx/fetcher.py:25]: Is this also supposed to be "V2"?

```suggestion
         'b847f053fc694d55d935c0be0e5268f7'), # V2 (27.09.2022)
```

Comment (frheault) [reply trx/fetcher.py:25]: There was no DSI.zip before, so this is the first version.

Comment (arokem) [reply trx/fetcher.py:25]: Gotcha. Sounds good!

**Pull request merged**